### ISSUE TITLE
[2] Can't restore PVCs with velero into restricted namespace

### ISSUE THREAD

**Issue by davidumea** [elastisys/compliantkubernetes-apps#1766]
**Describe the bug**:

When restoring a PVC + pod with velero, velero has to run an init container on the pod. This container gets blocked by our constraints when restoring into a restricted namespace.

```bash
  Warning  Failed  3m11s (x12 over 5m12s)  kubelet  Error: container has runAsNonRoot and image has non-numeric user (nonroot), cannot verify user is non-root (pod: "pod-name", container: restore-wait)
```

**To Reproduce**:

Steps to reproduce the behaviour:

1. Create a pod and PVC in a restricted subnamespace
1. Create a velero backup
1. Create a new restricted subnamespace via HNC
1. Restore a pod, persistentvolume and a persistentvolumeclaim from a backup into the new subnamespace
    - `velero restore create pvc-restore --from-backup <backup-name> --include-namespaces <old-ns> --include-resources pods,persistentvolumes,persistentvolumeclaims --namespace-mappings <old-ns>:<new-ns> --allow-partially-failed`
1. See that the velero init container `restore-wait` is not able to run, so the original container can not start either

**Expected behaviour**:

The velero init container should be able to run in a restricted namespace.

**Version (add all relevant versions)**:

- Compliant Kubernetes apps version v0.31

**Comment by simonklb** [comment]
Looks like this is still no fixed upstream and apparently it's not a simple fix as it has been open for quite a while: https://github.com/vmware-tanzu/velero/issues/2437

There is a fix in the OpenShift fork: https://github.com/openshift/velero but moving to that would probably need to be a separate issue.
